Slf4j logs are not showing in tests

The Slf4j logger is used in many places, such as in `ProgramExecutionSupport` https://github.com/enso-org/enso/blob/6cdbac66ea9cb98b7dbd9f93ba09be73b4737187/engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala#L62

## Issue

When running tests, the log statements produced by the `org.slf4j.LoggerFactory.getLogger(this.getClass)` are not visible. It makes debugging more difficult.

## How to reproduce

```scala
class Slf4jTest extends org.scalatest.flatspec.AnyFlatSpec {

  private val logger = org.slf4j.LoggerFactory.getLogger(this.getClass)

  it should "boo" in {
    logger.error("Boo!")
  }
}
```

The test will produce no output in SBT console:
`
```
sbt:enso> runtime-integration-tests/testOnly *.Slf4jTest
```

I've added the test as https://github.com/JaroslavTulach/enso/commit/9b24af9cc3ac84f3d6c7c6422eb547b6744ff57f and yes it doesn't log, when running:
```
sbt:enso> runtime-integration-tests/testOnly *Slf4jTest
[info] [FrgaalJavaCompiler] compiling 1 Java sources to /home/devel/NetBeansProjects/enso/enso/lib/scala/version-output/target/scala-2.13/classes ...
SLF4J(I): Attempting to load provider "org.enso.logging.service.logback.test.provider.TestLogProvider" specified via "slf4j.provider" system property
[info] Slf4jTest:
[info] - should boo (15 milliseconds)
[info] - should whoo when failing *** FAILED *** (1 millisecond)
[info]   java.lang.AssertionError: OK failure
[info]   at org.enso.runtime/org.enso.polyglot.test.Slf4jTest.$anonfun$new$2(Slf4jTest.scala:13)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.flatspec.AnyFlatSpecLike$$anon$5.apply(AnyFlatSpecLike.scala:1832)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[info]   at org.scalatest.flatspec.AnyFlatSpec.withFixture(AnyFlatSpec.scala:1686)
[info]   ...
[info] Run completed in 1 second, 36 milliseconds.
[info] Total number of tests run: 2
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 1, canceled 0, ignored 0, pending 0
[info] *** 1 TEST FAILED ***
[error] Failed tests:
[error]         org.enso.polyglot.test.Slf4jTest
```
- but wasn't there some @hubertp PR fixing logging in Scala tests under some circumstances?
- that was probably #13721 - but that PR hasn't been integrated

--- a/engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeAsyncCommandsTest.scala
+++ b/engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeAsyncCommandsTest.scala
@@ -31,7 +31,8 @@ class RuntimeAsyncCommandsTest
     with Matchers
     with BeforeAndAfterEach
     with DebugSpec
-    with FlakySpec {
+    with FlakySpec
+    with org.enso.testkit.ReportLogsOnFailure {
 
   // === Test Utilities =======================================================
 

--- a/engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeErrorsTest.scala
+++ b/engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeErrorsTest.scala
@@ -23,7 +23,8 @@ import java.util.logging.ConsoleHandler
 class RuntimeErrorsTest
     extends AnyFlatSpec
     with Matchers
-    with BeforeAndAfterEach {
+    with BeforeAndAfterEach
+    with org.enso.testkit.ReportLogsOnFailure {
 
   // === Test Utilities =======================================================
 

--- a/engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala
+++ b/engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala
@@ -26,7 +26,8 @@ import java.util.UUID
 class RuntimeServerTest
     extends AnyFlatSpec
     with Matchers
-    with BeforeAndAfterEach {
+    with BeforeAndAfterEach
+    with org.enso.testkit.ReportLogsOnFailure {
 
   // === Test Utilities =======================================================
 

--- a/engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala
+++ b/engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala
@@ -19,7 +19,10 @@ import java.util.UUID
 import java.util.logging.Level
 
 @scala.annotation.nowarn("msg=multiarg infix syntax")
-class RuntimeVisualizationsTest extends AnyFlatSpec with Matchers {
+class RuntimeVisualizationsTest
+    extends AnyFlatSpec
+    with Matchers
+    with org.enso.testkit.ReportLogsOnFailure {
 
   // === Test Utilities =======================================================
 

--- a/lib/scala/logging-utils/src/main/java/org/enso/logger/ObservedMessage.java
+++ b/lib/scala/logging-utils/src/main/java/org/enso/logger/ObservedMessage.java
@@ -115,6 +115,11 @@ public final class ObservedMessage {
     return formattedMsg.get();
   }
 
+  @Override
+  public String toString() {
+    return "[" + level + "] [" + instant + "] " + getFormattedMessage();
+  }
+
   /**
    * Provides <em>observing capabilities</em> to used logging infrastructure. Different
    * implementations of {@link Logger} have different ways of being <em>observed</em>. If one is

--- a/lib/scala/testkit/src/main/scala/org/enso/testkit/ReportLogsOnFailure.scala
+++ b/lib/scala/testkit/src/main/scala/org/enso/testkit/ReportLogsOnFailure.scala
@@ -1,42 +1,21 @@
 package org.enso.testkit
 
-import org.scalatest.{Args, Failed, Outcome, Status, TestSuite}
+import org.scalatest.{Failed, Outcome, TestSuite}
 import org.enso.logger.ObservedMessage
+import org.slf4j.LoggerFactory
+import org.slf4j.Logger
 
 trait ReportLogsOnFailure extends TestSuite {
 
-  abstract override protected def runTest(
-    testName: String,
-    args: Args
-  ): Status = {
-    val log    = org.slf4j.LoggerFactory.getLogger("org")
-    val ps     = System.out
-    val arr    = new java.util.ArrayList[ObservedMessage]()
-    val handle = ObservedMessage.observe(log, arr.add(_))
-    try {
-      super.runTest(testName, args)
-    } catch {
-      case e: Throwable =>
-        arr.forEach {
-          ps.println(_)
-        }
-        throw e
-    } finally {
-      arr.clear()
-      handle.close()
-    }
-  }
-
   abstract override def withFixture(test: NoArgTest): Outcome = {
-    val log    = org.slf4j.LoggerFactory.getLogger("org")
-    val ps     = System.out
+    val log    = LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME)
     val arr    = new java.util.ArrayList[ObservedMessage]()
     val handle = ObservedMessage.observe(log, arr.add(_))
     try {
       super.withFixture(test) match {
         case outcome @ Failed(_) =>
           arr.forEach {
-            ps.println(_)
+            System.out.println(_)
           }
           outcome
         case outcome =>

--- a/lib/scala/testkit/src/main/scala/org/enso/testkit/ReportLogsOnFailure.scala
+++ b/lib/scala/testkit/src/main/scala/org/enso/testkit/ReportLogsOnFailure.scala
@@ -1,51 +1,50 @@
 package org.enso.testkit
 
-import ch.qos.logback.classic.spi.ILoggingEvent
-import ch.qos.logback.core.Appender
-import org.enso.logging.service.logback.MemoryAppender
 import org.scalatest.{Args, Failed, Outcome, Status, TestSuite}
-import org.slf4j.{Logger, LoggerFactory}
+import org.enso.logger.ObservedMessage
 
 trait ReportLogsOnFailure extends TestSuite {
-
-  private lazy val appender: Appender[ILoggingEvent] = {
-    val ctx = LoggerFactory.getILoggerFactory;
-    val rootLogger = ctx
-      .getLogger(Logger.ROOT_LOGGER_NAME)
-      .asInstanceOf[ch.qos.logback.classic.Logger]
-    rootLogger.getAppender(MemoryAppender.NAME)
-  }
 
   abstract override protected def runTest(
     testName: String,
     args: Args
   ): Status = {
-    appender match {
-      case memoryAppender: MemoryAppender =>
-        memoryAppender.stopForwarding()
-        super.runTest(testName, args)
-      case _ =>
-        super.runTest(testName, args)
+    val log    = org.slf4j.LoggerFactory.getLogger("org")
+    val ps     = System.out
+    val arr    = new java.util.ArrayList[ObservedMessage]()
+    val handle = ObservedMessage.observe(log, arr.add(_))
+    try {
+      super.runTest(testName, args)
+    } catch {
+      case e: Throwable =>
+        arr.forEach {
+          ps.println(_)
+        }
+        throw e
+    } finally {
+      arr.clear()
+      handle.close()
     }
-
   }
 
   abstract override def withFixture(test: NoArgTest): Outcome = {
-    appender match {
-      case memoryAppender: MemoryAppender =>
-        try {
-          super.withFixture(test) match {
-            case outcome @ Failed(_) =>
-              memoryAppender.flush()
-              outcome
-            case outcome =>
-              outcome
+    val log    = org.slf4j.LoggerFactory.getLogger("org")
+    val ps     = System.out
+    val arr    = new java.util.ArrayList[ObservedMessage]()
+    val handle = ObservedMessage.observe(log, arr.add(_))
+    try {
+      super.withFixture(test) match {
+        case outcome @ Failed(_) =>
+          arr.forEach {
+            ps.println(_)
           }
-        } finally {
-          memoryAppender.reset()
-        }
-      case _ =>
-        super.withFixture(test)
+          outcome
+        case outcome =>
+          outcome
+      }
+    } finally {
+      arr.clear()
+      handle.close()
     }
   }
 

